src/main/scala/org/broadinstitute/dsde/rawls/expressions/SlickExpressionParsing.scala

```diff
@@ -32,10 +37,11 @@ trait SlickExpressionParser extends JavaTokenParsers {
 
   // converts the incoming PipeType into the appropriate db action
   // PipeType may be None when there is no pipeline (e.g. literals, workspace attributes)
-  type FinalFunc = (SlickExpressionContext, Option[PipeType]) => ReadAction[Iterable[Any]]
+  // Returns a map of entity names to an iterable of the expression result
+  type FinalFunc = (SlickExpressionContext, Option[PipeType]) => ReadAction[Map[String, Iterable[Any]]]
 
-  // a query that results in entity records and a sort ordering
-  type PipeType = Query[EntityTable, EntityRecord, Seq]
+  // a query that results in the root entity's name, entity records and a sort ordering
+  type PipeType = Query[(Rep[String], EntityTable), (String, EntityRecord), Seq]
```

Comment:
Get used to this dumb tuple. It's a tuple of the root entity for this expression (from `EXPREVAL_TABLE`) plus whatever entity/attribute we've worked our way along the expression chain to.